src/http_handlers/metrics.hpp

```diff
@@ -50,17 +54,85 @@ class MetricsService {
       : db_(data->db), interpreter_context_(data->interpreter_context), interpreter_(data->interpreter_context) {}
   MetricsResponse GetMetrics() {
     auto info = db_->GetInfo();
+
     return MetricsResponse{.vertex_count = info.vertex_count,
                            .edge_count = info.edge_count,
                            .average_degree = info.average_degree,
                            .memory_usage = info.memory_usage,
-                           .disk_usage = info.disk_usage};
+                           .disk_usage = info.disk_usage,
+                           .event_counters = GetEventCounters(),
+                           .event_gauges = GetEventGauges(),
+                           .event_histograms = GetEventHistograms()};
+  }
+
+  nlohmann::json AsJson(MetricsResponse response) {
+    auto metrics_response = nlohmann::json();
+    const auto *general_type = "General";
+
+    metrics_response[general_type]["vertex_count"] = response.vertex_count;
+    metrics_response[general_type]["edge_count"] = response.edge_count;
+    metrics_response[general_type]["average_degree"] = response.average_degree;
+    metrics_response[general_type]["memory_usage"] = response.memory_usage;
+    metrics_response[general_type]["disk_usage"] = response.disk_usage;
+
+    for (const auto &[name, type, value] : response.event_counters) {
+      metrics_response[type][name] = value;
+    }
+
+    for (const auto &[name, type, value] : response.event_gauges) {
+      metrics_response[type][name] = value;
+    }
+
+    for (const auto &[name, type, value] : response.event_histograms) {
+      metrics_response[type][name] = value;
+    }
+
+    return metrics_response;
   }
 
  private:
   const storage::Storage *db_;
   query::InterpreterContext *interpreter_context_;
   query::Interpreter interpreter_;
+
+  auto GetEventCounters() {
+    std::vector<std::tuple<std::string, std::string, uint64_t>> event_counters(memgraph::metrics::CounterEnd());
+
+    for (auto i = 0; i < memgraph::metrics::CounterEnd(); i++) {
+      event_counters.emplace_back(memgraph::metrics::GetCounterName(i), memgraph::metrics::GetCounterType(i),
+                                  memgraph::metrics::global_counters[i].load(std::memory_order_relaxed));
+    }
+
+    return event_counters;
+  }
+
+  auto GetEventGauges() {
+    std::vector<std::tuple<std::string, std::string, uint64_t>> event_gauges(memgraph::metrics::GaugeEnd());
+
+    for (auto i = 0; i < memgraph::metrics::GaugeEnd(); i++) {
+      event_gauges.emplace_back(memgraph::metrics::GetGaugeName(i), memgraph::metrics::GetGaugeType(i),
+                                memgraph::metrics::global_gauges[i].load(std::memory_order_seq_cst));
+    }
+
+    return event_gauges;
+  }
+
```

Comment:
Since you are doing load here, I think you can use `memory_order_acquire` as we are not reading anyting and `memory_order_seq_cst` includes reading too. Also, in function GetEventCounters I would also use `memory_order_acquire` instead of `memory_order_relaxed`